# Austin Animal Shelter Analysis

## Introduction

Austin Animal Centre is the municipal shelter for the City of Austin which provides shelter to thousands of animals annually as well as providing animal protection and pet resource services.

Their goal is to place all adoptable animals in forever homes through adoption, foster care, or rescue partner groups.


### About this analysis


This analysis will revolve around two csv files - intakes and outcomes - that contain personal and medical information about each animal that has entered and left the shelter.

### The Aim

The aim is to uncover the relationship between different animals' condition and characteristics when they have entered the shelter, and the results of their outcome (Leaving the shelter) taking in consideration their age, breed, the length of their stay, and the reason of intake and outcome. As well as come to a conclusion of whether it is possible to predict the outcome results of the other animals based off the previous results. Also, to measure how well the shelter is doing on basis of finding safer homes for these sheltered animals.

## Problem Statement

Since 2013 up to today, Austin Animal Center has been responsible of rescuing 172,647 different animals and found a forever home for 111,076. This means that only 64.3% of the animals rescued were adopted in the past 12 years.  And while the majority of rescued animals were rehomed or returned, about 6% had to be euthanized. To boost adoptions and reduce shelter stay times, it’s crucial to understand these trends, remove adoption barriers, and actively engage the community.

## Objectives

What intake types and conditions are most likely to lead to positive outcomes?

Which animal characteristics are most associated with longer shelter stays?

How do seasonal trends affect intake and outcome volumes throughout the year?

What is the average length of stay by outcome type, and how can it be reduced for less desirable outcomes?

Can we predict an animal’s outcome at the time of intake using available data?

## Target Audience

1- Austin Animal Shelter

2- Possible Adopter

3- Non-Profit organizations

4- Other Shelters

5- Possible Investors

## Datasets Used

https://data.austintexas.gov/browse?q=animal&sortBy=relevance&pageSize=20&page=1

Intakes

Outcomes

## Data Handeling

In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
intakes = pd.read_csv("Austin_Animal_Center_Intakes_20250408.csv")
outcomes = pd.read_csv("Austin_Animal_Center_Outcomes_20250408.csv")

In [3]:
#Sorting the Animal ID to check if the IDs on both files are similar
intakes.sort_values (by = "Animal ID" , ascending = True)

Animal ID    Name                DateTime      MonthYear  \
5647     A006100   Scamp  12/19/2014 10:21:00 AM  December 2014   
26429    A006100   Scamp  12/07/2017 02:07:00 PM  December 2017   
119397   A006100   Scamp  03/07/2014 02:26:00 PM     March 2014   
92797    A047759    Oreo  04/02/2014 03:55:00 PM     April 2014   
126267   A134067  Bandit  11/16/2013 09:02:00 AM  November 2013   
...          ...     ...                     ...            ...   
172901   A928084     NaN  04/07/2025 02:18:00 PM     April 2025   
172899   A928085   Ollie  04/07/2025 02:16:00 PM     April 2025   
172905   A928093     NaN  04/07/2025 03:34:00 PM     April 2025   
172903   A928094     NaN  04/07/2025 01:48:00 PM     April 2025   
172909   A928099     NaN  04/07/2025 07:03:00 PM     April 2025   

                                       Found Location      Intake Type  \
5647                8700 Research Blvd in Austin (TX)    Public Assist   
26429   Colony Creek And Hunters Trace in Austin (TX)            Stray   
119397                   8700 Research in Austin (TX)    Public Assist   
92797                                     Austin (TX)  Owner Surrender   
126267             12034 Research Blvd in Austin (TX)    Public Assist   
...                                               ...              ...   
172901                                    Austin (TX)  Owner Surrender   
172899                                    Austin (TX)  Owner Surrender   
172905                2108 Ferguson Ln in Austin (TX)            Stray   
172903                2108 Ferguson Ln in Austin (TX)            Stray   
172909        Crestview And Park Dr in Jonestown (TX)            Stray   

       Intake Condition Animal Type Sex upon Intake Age upon Intake  \
5647             Normal         Dog   Neutered Male         7 years   
26429            Normal         Dog   Neutered Male        10 years   
119397           Normal         Dog   Neutered Male         6 years   
92797            Normal         Dog   Neutered Male        10 years   
126267          Injured         Dog   Neutered Male        16 years   
...                 ...         ...             ...             ...   
172901             Sick         Dog   Intact Female        5 months   
172899           Normal         Dog   Neutered Male         2 years   
172905           Normal         Dog     Intact Male        4 months   
172903           Normal         Dog     Intact Male        4 months   
172909           Normal         Dog         Unknown         7 years   

                              Breed         Color  
5647           Spinone Italiano Mix  Yellow/White  
26429          Spinone Italiano Mix  Yellow/White  
119397         Spinone Italiano Mix  Yellow/White  
92797                     Dachshund      Tricolor  
126267            Shetland Sheepdog   Brown/White  
...                             ...           ...  
172901           Great Pyrenees Mix         White  
172899       Labrador Retriever Mix   White/Black  
172905      Chihuahua Shorthair Mix      Tricolor  
172903      Chihuahua Shorthair Mix      Tricolor  
172909  Labrador Retriever/Pit Bull    White/Gray  

[172911 rows x 12 columns]

In [4]:
outcomes.sort_values (by = "Animal ID" , ascending = True)

Animal ID        Name          DateTime MonthYear Date of Birth  \
74201    A006100       Scamp  12/20/2014 16:35    Dec-14      7/9/2007   
150491   A006100       Scamp    3/8/2014 17:10    Mar-14      7/9/2007   
132573   A006100       Scamp    12/7/2017 0:00    Dec-17      7/9/2007   
51216    A047759        Oreo    4/7/2014 15:12    Apr-14      4/2/2004   
106445   A134067      Bandit  11/16/2013 11:54    Nov-13    10/16/1997   
...          ...         ...               ...       ...           ...   
172940   A927961        Kira    4/4/2025 19:01    Apr-25      4/4/2023   
172981   A927962         NaN    4/6/2025 12:52    Apr-25      6/4/2024   
4673     A927988     Unknown     4/6/2025 8:42    Apr-25      4/5/2023   
172962   A927996     Unknown    4/5/2025 15:32    Apr-25      4/5/2017   
173003   A928024  Pinto Bean    4/7/2025 11:41    Apr-25      4/6/2019   

           Outcome Type Outcome Subtype Animal Type Sex upon Outcome  \
74201   Return to Owner             NaN         Dog    Neutered Male   
150491  Return to Owner             NaN         Dog    Neutered Male   
132573  Return to Owner             NaN         Dog    Neutered Male   
51216          Transfer         Partner         Dog    Neutered Male   
106445  Return to Owner             NaN         Dog    Neutered Male   
...                 ...             ...         ...              ...   
172940  Return to Owner             NaN         Dog    Spayed Female   
172981  Return to Owner             NaN         Cat          Unknown   
4673         Euthanasia     Rabies Risk       Other          Unknown   
172962       Euthanasia       Suffering         Cat    Spayed Female   
173003  Return to Owner             NaN         Dog    Neutered Male   

       Age upon Outcome                 Breed              Color  
74201           7 years  Spinone Italiano Mix       Yellow/White  
150491          6 years  Spinone Italiano Mix       Yellow/White  
132573         10 years  Spinone Italiano Mix       Yellow/White  
51216          10 years             Dachshund           Tricolor  
106445         16 years     Shetland Sheepdog        Brown/White  
...                 ...                   ...                ...  
172940          2 years              Pit Bull        Black/White  
172981        10 months    Domestic Shorthair       Orange Tabby  
4673            2 years                   Bat              Brown  
172962              NaN    Domestic Shorthair         Gray Tabby  
173003          6 years           Whippet Mix  Brown Merle/White  

[173025 rows x 12 columns]

In [5]:
#Checking the data types of each column to make sure that they are fit 
intakes.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 172911 entries, 0 to 172910
Data columns (total 12 columns):
 #   Column            Non-Null Count   Dtype 
---  ------            --------------   ----- 
 0   Animal ID         172911 non-null  object
 1   Name              123299 non-null  object
 2   DateTime          172911 non-null  object
 3   MonthYear         172911 non-null  object
 4   Found Location    172911 non-null  object
 5   Intake Type       172911 non-null  object
 6   Intake Condition  172911 non-null  object
 7   Animal Type       172911 non-null  object
 8   Sex upon Intake   172909 non-null  object
 9   Age upon Intake   172910 non-null  object
 10  Breed             172911 non-null  object
 11  Color             172911 non-null  object
dtypes: object(12)
memory usage: 15.8+ MB


In [6]:
outcomes.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 173025 entries, 0 to 173024
Data columns (total 12 columns):
 #   Column            Non-Null Count   Dtype 
---  ------            --------------   ----- 
 0   Animal ID         173025 non-null  object
 1   Name              123502 non-null  object
 2   DateTime          173025 non-null  object
 3   MonthYear         173025 non-null  object
 4   Date of Birth     173025 non-null  object
 5   Outcome Type      172979 non-null  object
 6   Outcome Subtype   79275 non-null   object
 7   Animal Type       173025 non-null  object
 8   Sex upon Outcome  173023 non-null  object
 9   Age upon Outcome  173020 non-null  object
 10  Breed             173025 non-null  object
 11  Color             173025 non-null  object
dtypes: object(12)
memory usage: 15.8+ MB


In [7]:
#Asigning the correct data type for each
intakes["DateTime"] = pd.to_datetime(intakes["DateTime"])
outcomes["DateTime"] = pd.to_datetime(outcomes["DateTime"])
outcomes["Date of Birth"] = pd.to_datetime(outcomes["Date of Birth"])

C:\Users\shahad\AppData\Local\Temp\ipykernel_24668\1998269207.py:2: UserWarning: Could not infer format, so each element will be parsed individually, falling back to `dateutil`. To ensure parsing is consistent and as-expected, please specify a format.
  intakes["DateTime"] = pd.to_datetime(intakes["DateTime"])


In [8]:
outcomes.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 173025 entries, 0 to 173024
Data columns (total 12 columns):
 #   Column            Non-Null Count   Dtype         
---  ------            --------------   -----         
 0   Animal ID         173025 non-null  object        
 1   Name              123502 non-null  object        
 2   DateTime          173025 non-null  datetime64[ns]
 3   MonthYear         173025 non-null  object        
 4   Date of Birth     173025 non-null  datetime64[ns]
 5   Outcome Type      172979 non-null  object        
 6   Outcome Subtype   79275 non-null   object        
 7   Animal Type       173025 non-null  object        
 8   Sex upon Outcome  173023 non-null  object        
 9   Age upon Outcome  173020 non-null  object        
 10  Breed             173025 non-null  object        
 11  Color             173025 non-null  object        
dtypes: datetime64[ns](2), object(10)
memory usage: 15.8+ MB


In [9]:
intakes.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 172911 entries, 0 to 172910
Data columns (total 12 columns):
 #   Column            Non-Null Count   Dtype         
---  ------            --------------   -----         
 0   Animal ID         172911 non-null  object        
 1   Name              123299 non-null  object        
 2   DateTime          172911 non-null  datetime64[ns]
 3   MonthYear         172911 non-null  object        
 4   Found Location    172911 non-null  object        
 5   Intake Type       172911 non-null  object        
 6   Intake Condition  172911 non-null  object        
 7   Animal Type       172911 non-null  object        
 8   Sex upon Intake   172909 non-null  object        
 9   Age upon Intake   172910 non-null  object        
 10  Breed             172911 non-null  object        
 11  Color             172911 non-null  object        
dtypes: datetime64[ns](1), object(11)
memory usage: 15.8+ MB


In [10]:
intakes.head()

Animal ID          Name            DateTime     MonthYear  \
0   A786884        *Brock 2019-01-03 16:19:00  January 2019   
1   A706918         Belle 2015-07-05 12:59:00     July 2015   
2   A724273       Runster 2016-04-14 18:43:00    April 2016   
3   A665644           NaN 2013-10-21 07:59:00  October 2013   
4   A857105  Johnny Ringo 2022-05-12 00:23:00      May 2022   

                        Found Location    Intake Type Intake Condition  \
0  2501 Magin Meadow Dr in Austin (TX)          Stray           Normal   
1     9409 Bluegrass Dr in Austin (TX)          Stray           Normal   
2   2818 Palomino Trail in Austin (TX)          Stray           Normal   
3                          Austin (TX)          Stray             Sick   
4   4404 Sarasota Drive in Austin (TX)  Public Assist           Normal   

  Animal Type Sex upon Intake Age upon Intake                     Breed  \
0         Dog   Neutered Male         2 years                Beagle Mix   
1         Dog   Spayed Female         8 years  English Springer Spaniel   
2         Dog     Intact Male       11 months               Basenji Mix   
3         Cat   Intact Female         4 weeks    Domestic Shorthair Mix   
4         Cat   Neutered Male         2 years        Domestic Shorthair   

          Color  
0      Tricolor  
1   White/Liver  
2   Sable/White  
3        Calico  
4  Orange Tabby

# Exploring the Dataframes

In [11]:
intakes.columns

Index(['Animal ID', 'Name', 'DateTime', 'MonthYear', 'Found Location',
       'Intake Type', 'Intake Condition', 'Animal Type', 'Sex upon Intake',
       'Age upon Intake', 'Breed', 'Color'],
      dtype='object')

In [12]:
outcomes.columns

Index(['Animal ID', 'Name', 'DateTime', 'MonthYear', 'Date of Birth',
       'Outcome Type', 'Outcome Subtype', 'Animal Type', 'Sex upon Outcome',
       'Age upon Outcome', 'Breed', 'Color'],
      dtype='object')

In [13]:
intakes.isnull().sum()

Animal ID               0
Name                49612
DateTime                0
MonthYear               0
Found Location          0
Intake Type             0
Intake Condition        0
Animal Type             0
Sex upon Intake         2
Age upon Intake         1
Breed                   0
Color                   0
dtype: int64

In [14]:
outcomes.isnull().sum()

Animal ID               0
Name                49523
DateTime                0
MonthYear               0
Date of Birth           0
Outcome Type           46
Outcome Subtype     93750
Animal Type             0
Sex upon Outcome        2
Age upon Outcome        5
Breed                   0
Color                   0
dtype: int64

In [15]:
intakes.duplicated().sum()

35

In [16]:
outcomes.duplicated().sum()

25

In [17]:
intakes["Intake Type"].unique()

array(['Stray', 'Public Assist', 'Owner Surrender', 'Abandoned',
       'Wildlife', 'Euthanasia Request'], dtype=object)

In [18]:
outcomes["Outcome Type"].unique()

array(['Adoption', 'Rto-Adopt', 'Euthanasia', 'Transfer',
       'Return to Owner', 'Died', 'Disposal', 'Missing', nan, 'Relocate',
       'Stolen', 'Lost'], dtype=object)

In [19]:
outcomes["Animal Type"].unique()

array(['Cat', 'Dog', 'Other', 'Bird', 'Livestock'], dtype=object)

# Cleaning the Data


After noticing the date of birth column is missing from the intakes table, we will create a new dataframe that will include the unique Animal ID with their dates of birth

In [20]:
DOB = outcomes[["Animal ID" , "Date of Birth"]]

In [21]:
DOB.head()

Animal ID Date of Birth
0   A882831    2023-03-25
1   A794011    2017-05-02
2   A776359    2017-07-12
3   A821648    2019-08-16
4   A720371    2015-10-08

In [22]:
DOB.duplicated().sum()

17471

In [23]:
DOB = DOB.drop_duplicates(subset="Animal ID")

In [24]:
DOB.duplicated().sum()

0

Then we will have to calculate the estimated stay per animal.

In [25]:
intakes.head()

Animal ID          Name            DateTime     MonthYear  \
0   A786884        *Brock 2019-01-03 16:19:00  January 2019   
1   A706918         Belle 2015-07-05 12:59:00     July 2015   
2   A724273       Runster 2016-04-14 18:43:00    April 2016   
3   A665644           NaN 2013-10-21 07:59:00  October 2013   
4   A857105  Johnny Ringo 2022-05-12 00:23:00      May 2022   

                        Found Location    Intake Type Intake Condition  \
0  2501 Magin Meadow Dr in Austin (TX)          Stray           Normal   
1     9409 Bluegrass Dr in Austin (TX)          Stray           Normal   
2   2818 Palomino Trail in Austin (TX)          Stray           Normal   
3                          Austin (TX)          Stray             Sick   
4   4404 Sarasota Drive in Austin (TX)  Public Assist           Normal   

  Animal Type Sex upon Intake Age upon Intake                     Breed  \
0         Dog   Neutered Male         2 years                Beagle Mix   
1         Dog   Spayed Female         8 years  English Springer Spaniel   
2         Dog     Intact Male       11 months               Basenji Mix   
3         Cat   Intact Female         4 weeks    Domestic Shorthair Mix   
4         Cat   Neutered Male         2 years        Domestic Shorthair   

          Color  
0      Tricolor  
1   White/Liver  
2   Sable/White  
3        Calico  
4  Orange Tabby

In [26]:
intakes = intakes.drop(columns = "MonthYear")

In [27]:
def extract_months(age_str):
    if pd.isnull(age_str):
        return None
    try:
        number, unit = age_str.strip().lower().split()[:2]
        number = float(number)
        if 'year' in unit:
            return number * 12
        elif 'month' in unit:
            return number
        elif 'week' in unit:
            return number / 4
        elif 'day' in unit:
            return number / 30
        else:
            return None
    except:
        return None

# Apply to dataset
intakes["Age in Months"] = intakes["Age upon Intake"].apply(extract_months)
intakes.head()


Animal ID          Name            DateTime  \
0   A786884        *Brock 2019-01-03 16:19:00   
1   A706918         Belle 2015-07-05 12:59:00   
2   A724273       Runster 2016-04-14 18:43:00   
3   A665644           NaN 2013-10-21 07:59:00   
4   A857105  Johnny Ringo 2022-05-12 00:23:00   

                        Found Location    Intake Type Intake Condition  \
0  2501 Magin Meadow Dr in Austin (TX)          Stray           Normal   
1     9409 Bluegrass Dr in Austin (TX)          Stray           Normal   
2   2818 Palomino Trail in Austin (TX)          Stray           Normal   
3                          Austin (TX)          Stray             Sick   
4   4404 Sarasota Drive in Austin (TX)  Public Assist           Normal   

  Animal Type Sex upon Intake Age upon Intake                     Breed  \
0         Dog   Neutered Male         2 years                Beagle Mix   
1         Dog   Spayed Female         8 years  English Springer Spaniel   
2         Dog     Intact Male       11 months               Basenji Mix   
3         Cat   Intact Female         4 weeks    Domestic Shorthair Mix   
4         Cat   Neutered Male         2 years        Domestic Shorthair   

          Color  Age in Months  
0      Tricolor           24.0  
1   White/Liver           96.0  
2   Sable/White           11.0  
3        Calico            1.0  
4  Orange Tabby           24.0

In [28]:
intakes = intakes.drop(columns = "Age upon Intake")

In [29]:
intakes["Sex upon Intake"] = intakes["Sex upon Intake"].fillna("Unknown Unknown")

intakes[["Reproductive Status", "Gender"]] = intakes["Sex upon Intake"].str.strip().str.split(' ', n=1, expand=True)

In [30]:
intakes.head()

Animal ID          Name            DateTime  \
0   A786884        *Brock 2019-01-03 16:19:00   
1   A706918         Belle 2015-07-05 12:59:00   
2   A724273       Runster 2016-04-14 18:43:00   
3   A665644           NaN 2013-10-21 07:59:00   
4   A857105  Johnny Ringo 2022-05-12 00:23:00   

                        Found Location    Intake Type Intake Condition  \
0  2501 Magin Meadow Dr in Austin (TX)          Stray           Normal   
1     9409 Bluegrass Dr in Austin (TX)          Stray           Normal   
2   2818 Palomino Trail in Austin (TX)          Stray           Normal   
3                          Austin (TX)          Stray             Sick   
4   4404 Sarasota Drive in Austin (TX)  Public Assist           Normal   

  Animal Type Sex upon Intake                     Breed         Color  \
0         Dog   Neutered Male                Beagle Mix      Tricolor   
1         Dog   Spayed Female  English Springer Spaniel   White/Liver   
2         Dog     Intact Male               Basenji Mix   Sable/White   
3         Cat   Intact Female    Domestic Shorthair Mix        Calico   
4         Cat   Neutered Male        Domestic Shorthair  Orange Tabby   

   Age in Months Reproductive Status  Gender  
0           24.0            Neutered    Male  
1           96.0              Spayed  Female  
2           11.0              Intact    Male  
3            1.0              Intact  Female  
4           24.0            Neutered    Male

In [31]:
intakes = intakes.drop(columns = ["Sex upon Intake"])

In [32]:
intakes["Intake Condition"].unique()

array(['Normal', 'Sick', 'Injured', 'Pregnant', 'Neonatal', 'Nursing',
       'Aged', 'Unknown', 'Med Attn', 'Medical', 'Other', 'Med Urgent',
       'Behavior', 'Feral', 'Parvo', 'Neurologic', 'Agonal', 'Space',
       'Panleuk', 'Congenital'], dtype=object)

In [33]:
condition_map = {
    'Normal': 'Healthy',
    'Pregnant': 'Healthy',
    'Nursing': 'Healthy',
    'Aged': 'Healthy',

    'Sick': 'Medical - Moderate',
    'Injured': 'Medical - Moderate',
    'Med Attn': 'Medical - Moderate',
    'Medical': 'Medical - Moderate',
    'Other': 'Medical - Moderate',

    'Med Urgent': 'Medical - Critical',
    'Parvo': 'Medical - Critical',
    'Panleuk': 'Medical - Critical',
    'Neurologic': 'Medical - Critical',
    'Agonal': 'Medical - Critical',
    'Congenital': 'Medical - Critical',

    'Behavior': 'Behavioral',
    'Feral': 'Behavioral',
    'Space': 'Behavioral',

    'Unknown': 'Unknown',
    'Neonatal': 'Unknown'
}

intakes["Condition Category"] = intakes["Intake Condition"].map(condition_map)

In [34]:
intakes.head()

Animal ID          Name            DateTime  \
0   A786884        *Brock 2019-01-03 16:19:00   
1   A706918         Belle 2015-07-05 12:59:00   
2   A724273       Runster 2016-04-14 18:43:00   
3   A665644           NaN 2013-10-21 07:59:00   
4   A857105  Johnny Ringo 2022-05-12 00:23:00   

                        Found Location    Intake Type Intake Condition  \
0  2501 Magin Meadow Dr in Austin (TX)          Stray           Normal   
1     9409 Bluegrass Dr in Austin (TX)          Stray           Normal   
2   2818 Palomino Trail in Austin (TX)          Stray           Normal   
3                          Austin (TX)          Stray             Sick   
4   4404 Sarasota Drive in Austin (TX)  Public Assist           Normal   

  Animal Type                     Breed         Color  Age in Months  \
0         Dog                Beagle Mix      Tricolor           24.0   
1         Dog  English Springer Spaniel   White/Liver           96.0   
2         Dog               Basenji Mix   Sable/White           11.0   
3         Cat    Domestic Shorthair Mix        Calico            1.0   
4         Cat        Domestic Shorthair  Orange Tabby           24.0   

  Reproductive Status  Gender  Condition Category  
0            Neutered    Male             Healthy  
1              Spayed  Female             Healthy  
2              Intact    Male             Healthy  
3              Intact  Female  Medical - Moderate  
4            Neutered    Male             Healthy

In [35]:
intakes = intakes.drop(columns = ["Intake Condition"])

In [36]:
intakes["Intake Type"].unique()

array(['Stray', 'Public Assist', 'Owner Surrender', 'Abandoned',
       'Wildlife', 'Euthanasia Request'], dtype=object)

In [75]:
intakes["Color"].unique()

array(['Yellow/White', 'Tricolor', 'Brown/White', 'Black/White',
       'Black/Gray', 'Gray/White', 'White/Black', 'Brown Tabby/White',
       'Black/Tan', 'Tan/Black', 'Black', 'Gold/Gold', 'Tortie Point',
       'Sable/White', 'Blue Merle/Tan', 'Blue Merle', 'Brown/Black',
       'Black/Tricolor', 'Black/Black', 'Tan', 'Black/Brown', 'Cream',
       'Red', 'Yellow', 'Gray Tabby/White', 'Brown Merle', 'White',
       'White/Brown', 'Tortie', 'White/Gray Tabby', 'Tan/Red',
       'Black/Brown Brindle', 'Black Smoke/White', 'Calico', 'Brown/Tan',
       'Orange Tabby/White', 'Blue Cream/Tortie', 'Lynx Point',
       'Brown/Buff', 'Red Merle', 'Brown', 'Torbie', 'Brown/Cream',
       'Blue/White', 'Red/White', 'Tan/White', 'Lilac Point', 'Blue',
       'Gold', 'Brown Tabby', 'Red/Black', 'Black Brindle',
       'Brown/Blue Merle', 'Cream/White', 'Red/Tan', 'Black/Red',
       'Chocolate', 'Brown Tabby/Agouti', 'Blue Point',
       'Black Tabby/White', 'Silver Tabby', 'Brown Brindle/White

In [78]:
# Define your base colors
base_colors = {
    "black", "white", "gray", "brown", "red", "tan", "cream", "blue", "buff", "fawn",
    "orange", "yellow", "silver", "gold", "chocolate", "green", "apricot", "pink",
    "ruddy", "liver", "lilac"
}

# Function to categorize each color
def categorize_color(color_name):
    color_name = str(color_name).lower().replace("/", " ")
    words = color_name.split()

    if color_name in ("unknown", "unspecified"):
        return "Unknown/Unusual"
    elif len(words) == 1 and words[0] in base_colors:
        return "Base Color"
    elif all(word in base_colors for word in words):
        return "Color Combination"
    elif any(word in base_colors for word in words):
        return "Descriptive Color"
    else:
        return "Unknown/Unusual"

# Apply function and create new column
intakes["Color Category"] = intakes["Color"].apply(categorize_color)

intakes.head()


Animal ID    Name         Intake Date  \
119397   A006100   Scamp 2014-03-07 14:26:00   
5647     A006100   Scamp 2014-12-19 10:21:00   
26429    A006100   Scamp 2017-12-07 14:07:00   
92797    A047759    Oreo 2014-04-02 15:55:00   
126267   A134067  Bandit 2013-11-16 09:02:00   

                                       Found Location      Intake Type  \
119397                   8700 Research in Austin (TX)    Public Assist   
5647                8700 Research Blvd in Austin (TX)    Public Assist   
26429   Colony Creek And Hunters Trace in Austin (TX)            Stray   
92797                                     Austin (TX)  Owner Surrender   
126267             12034 Research Blvd in Austin (TX)    Public Assist   

       Animal Type                 Breed         Color  Age in Months  \
119397         Dog  Spinone Italiano Mix  Yellow/White           72.0   
5647           Dog  Spinone Italiano Mix  Yellow/White           84.0   
26429          Dog  Spinone Italiano Mix  Yellow/White          120.0   
92797          Dog             Dachshund      Tricolor          120.0   
126267         Dog     Shetland Sheepdog   Brown/White          192.0   

       Reproductive Status Gender  Condition Category  Entry Index  \
119397            Neutered   Male             Healthy            1   
5647              Neutered   Male             Healthy            2   
26429             Neutered   Male             Healthy            3   
92797             Neutered   Male             Healthy            1   
126267            Neutered   Male  Medical - Moderate            1   

           Color Category  
119397  Color Combination  
5647    Color Combination  
26429   Color Combination  
92797     Unknown/Unusual  
126267  Color Combination

In [97]:
intakes["Found Location"].value_counts()

Found Location
Austin (TX)                                             31404
Travis (TX)                                              3774
Outside Jurisdiction                                     2073
7201 Levander Loop in Austin (TX)                        1574
Manor (TX)                                                757
                                                        ...  
11901 Green Grove Drive in Austin (TX)                      1
Tuscany Way And Ferguson Ln in Austin (TX)                  1
Hwy 29 And Parmer Ln in Manor (TX)                          1
E Riverside Dr And Pleasant Valley St in Austin (TX)        1
Crestview And Park Dr in Jonestown (TX)                     1
Name: count, Length: 69894, dtype: int64

After we are done from creating the DOB dataframe and the intakes file, we will move on to check the outcomes.

In [37]:
outcomes["Age upon Outcome"].isnull().sum()

5

In [38]:
outcomes["Date of Birth"].isnull().sum()

0

In [39]:
# Calculate age in days, then convert to years or months as needed
outcomes["Age in Days"] = (outcomes["DateTime"] - outcomes["Date of Birth"]).dt.days
outcomes["Age in Years"] = outcomes["Age in Days"] / 365.25
outcomes["Age in Months"] = outcomes["Age in Days"] / 30.44

In [40]:
outcomes.head()

Animal ID       Name            DateTime MonthYear Date of Birth  \
0   A882831  *Hamilton 2023-07-01 18:12:00    Jul-23    2023-03-25   
1   A794011      Chunk 2019-05-08 18:20:00    May-19    2017-05-02   
2   A776359      Gizmo 2018-07-18 16:02:00    Jul-18    2017-07-12   
3   A821648        NaN 2020-08-16 11:38:00    Aug-20    2019-08-16   
4   A720371      Moose 2016-02-13 17:59:00    Feb-16    2015-10-08   

  Outcome Type Outcome Subtype Animal Type Sex upon Outcome Age upon Outcome  \
0     Adoption             NaN         Cat    Neutered Male         3 months   
1    Rto-Adopt             NaN         Cat    Neutered Male          2 years   
2     Adoption             NaN         Dog    Neutered Male           1 year   
3   Euthanasia             NaN       Other          Unknown           1 year   
4     Adoption             NaN         Dog    Neutered Male         4 months   

                                Breed              Color  Age in Days  \
0              Domestic Shorthair Mix        Black/White           98   
1              Domestic Shorthair Mix  Brown Tabby/White          736   
2             Chihuahua Shorthair Mix        White/Brown          371   
3                             Raccoon               Gray          366   
4  Anatol Shepherd/Labrador Retriever               Buff          128   

   Age in Years  Age in Months  
0      0.268309       3.219448  
1      2.015058      24.178712  
2      1.015743      12.187911  
3      1.002053      12.023653  
4      0.350445       4.204993

In [41]:
outcomes = outcomes.drop(columns = ["Age in Days","Age in Years","Age upon Outcome"])

In [42]:
outcomes.head()

Animal ID       Name            DateTime MonthYear Date of Birth  \
0   A882831  *Hamilton 2023-07-01 18:12:00    Jul-23    2023-03-25   
1   A794011      Chunk 2019-05-08 18:20:00    May-19    2017-05-02   
2   A776359      Gizmo 2018-07-18 16:02:00    Jul-18    2017-07-12   
3   A821648        NaN 2020-08-16 11:38:00    Aug-20    2019-08-16   
4   A720371      Moose 2016-02-13 17:59:00    Feb-16    2015-10-08   

  Outcome Type Outcome Subtype Animal Type Sex upon Outcome  \
0     Adoption             NaN         Cat    Neutered Male   
1    Rto-Adopt             NaN         Cat    Neutered Male   
2     Adoption             NaN         Dog    Neutered Male   
3   Euthanasia             NaN       Other          Unknown   
4     Adoption             NaN         Dog    Neutered Male   

                                Breed              Color  Age in Months  
0              Domestic Shorthair Mix        Black/White       3.219448  
1              Domestic Shorthair Mix  Brown Tabby/White      24.178712  
2             Chihuahua Shorthair Mix        White/Brown      12.187911  
3                             Raccoon               Gray      12.023653  
4  Anatol Shepherd/Labrador Retriever               Buff       4.204993

In [43]:
outcomes["Sex upon Outcome"] = outcomes["Sex upon Outcome"].fillna("Unknown Unknown")

outcomes[["Reproductive Status", "Gender"]] = outcomes["Sex upon Outcome"].str.strip().str.split(' ', n=1, expand=True)

In [44]:
outcomes = outcomes.drop(columns = "Outcome Subtype")

In [45]:
outcomes.head()

Animal ID       Name            DateTime MonthYear Date of Birth  \
0   A882831  *Hamilton 2023-07-01 18:12:00    Jul-23    2023-03-25   
1   A794011      Chunk 2019-05-08 18:20:00    May-19    2017-05-02   
2   A776359      Gizmo 2018-07-18 16:02:00    Jul-18    2017-07-12   
3   A821648        NaN 2020-08-16 11:38:00    Aug-20    2019-08-16   
4   A720371      Moose 2016-02-13 17:59:00    Feb-16    2015-10-08   

  Outcome Type Animal Type Sex upon Outcome  \
0     Adoption         Cat    Neutered Male   
1    Rto-Adopt         Cat    Neutered Male   
2     Adoption         Dog    Neutered Male   
3   Euthanasia       Other          Unknown   
4     Adoption         Dog    Neutered Male   

                                Breed              Color  Age in Months  \
0              Domestic Shorthair Mix        Black/White       3.219448   
1              Domestic Shorthair Mix  Brown Tabby/White      24.178712   
2             Chihuahua Shorthair Mix        White/Brown      12.187911   
3                             Raccoon               Gray      12.023653   
4  Anatol Shepherd/Labrador Retriever               Buff       4.204993   

  Reproductive Status Gender  
0            Neutered   Male  
1            Neutered   Male  
2            Neutered   Male  
3             Unknown   None  
4            Neutered   Male

In [46]:
outcomes["Outcome Type"].unique()

array(['Adoption', 'Rto-Adopt', 'Euthanasia', 'Transfer',
       'Return to Owner', 'Died', 'Disposal', 'Missing', nan, 'Relocate',
       'Stolen', 'Lost'], dtype=object)

In [47]:
outcome_map = {
    'Adoption': 'Positive',
    'Rto-Adopt': 'Positive',
    'Transfer': 'Positive',
    'Return to Owner': 'Positive',
    'Euthanasia': 'Negative',
    'Died': 'Negative',
    'Disposal': 'Negative',
    'Stolen': 'Negative',
    'Missing': 'Neutral',
    'Lost': 'Neutral',
    'Relocate': 'Neutral'
}

outcomes["Outcome Category"] = outcomes["Outcome Type"].map(outcome_map)

In [48]:
outcomes["Outcome Type"].unique()

array(['Adoption', 'Rto-Adopt', 'Euthanasia', 'Transfer',
       'Return to Owner', 'Died', 'Disposal', 'Missing', nan, 'Relocate',
       'Stolen', 'Lost'], dtype=object)

In [49]:
outcomes["Outcome Type"].isnull().sum()

46

In [79]:
outcomes["Color Category"] = outcomes["Color"].apply(categorize_color)

In [84]:
outcomes = outcomes.drop(columns = "MonthYear")

In [85]:
outcomes.head()

Animal ID    Name        Outcome Date Date of Birth     Outcome Type  \
150491   A006100   Scamp 2014-03-08 17:10:00    2007-07-09  Return to Owner   
51216    A047759    Oreo 2014-04-07 15:12:00    2004-04-02         Transfer   
106445   A134067  Bandit 2013-11-16 11:54:00    1997-10-16  Return to Owner   
168829   A141142  Bettie 2013-11-17 11:40:00    1998-06-01  Return to Owner   
172531   A163459   Sasha 2014-11-14 19:28:00    1999-10-19  Return to Owner   

       Animal Type Sex upon Outcome                        Breed  \
150491         Dog    Neutered Male         Spinone Italiano Mix   
51216          Dog    Neutered Male                    Dachshund   
106445         Dog    Neutered Male            Shetland Sheepdog   
168829         Dog    Spayed Female  Labrador Retriever/Pit Bull   
172531         Dog    Intact Female      Miniature Schnauzer Mix   

               Color  Age in Months Reproductive Status  Gender  \
150491  Yellow/White      79.960578            Neutered    Male   
51216       Tricolor     120.137976            Neutered    Male   
106445   Brown/White     193.002628            Neutered    Male   
168829   Black/White     185.545335              Spayed  Female   
172531    Black/Gray     180.847569              Intact  Female   

       Outcome Category  Entry Index     Color Category  
150491         Positive            1  Color Combination  
51216          Positive            1    Unknown/Unusual  
106445         Positive            1  Color Combination  
168829         Positive            1  Color Combination  
172531         Positive            1  Color Combination

Now we will be calculating the estimated stay for each animal.
And to do so, we will create a new dataframe that will contain the Animal ID, Intake date, and the outcome date.

First off we will change the column names to be more suitable.

In [50]:
intakes.rename(columns = {"DateTime" : "Intake Date"}, inplace = True)

In [51]:
outcomes.rename(columns = {"DateTime" : "Outcome Date"}, inplace = True)

Then we will sort the values into ascending to prevent any mix ups.

In [52]:
intakes = intakes.sort_values(by=["Animal ID" , "Intake Date"])

In [53]:
outcomes = outcomes.sort_values(by = ["Animal ID" , "Outcome Date"])

Then we will set an index for both

In [54]:
intakes['Entry Index'] = intakes.groupby('Animal ID').cumcount() + 1

In [55]:
intakes.shape

(172911, 13)

In [56]:
outcomes['Entry Index'] = outcomes.groupby('Animal ID').cumcount() + 1

In [57]:
outcomes.shape

(173025, 15)

In [58]:
intakes

Animal ID    Name         Intake Date  \
119397   A006100   Scamp 2014-03-07 14:26:00   
5647     A006100   Scamp 2014-12-19 10:21:00   
26429    A006100   Scamp 2017-12-07 14:07:00   
92797    A047759    Oreo 2014-04-02 15:55:00   
126267   A134067  Bandit 2013-11-16 09:02:00   
...          ...     ...                 ...   
172901   A928084     NaN 2025-04-07 14:18:00   
172899   A928085   Ollie 2025-04-07 14:16:00   
172905   A928093     NaN 2025-04-07 15:34:00   
172903   A928094     NaN 2025-04-07 13:48:00   
172909   A928099     NaN 2025-04-07 19:03:00   

                                       Found Location      Intake Type  \
119397                   8700 Research in Austin (TX)    Public Assist   
5647                8700 Research Blvd in Austin (TX)    Public Assist   
26429   Colony Creek And Hunters Trace in Austin (TX)            Stray   
92797                                     Austin (TX)  Owner Surrender   
126267             12034 Research Blvd in Austin (TX)    Public Assist   
...                                               ...              ...   
172901                                    Austin (TX)  Owner Surrender   
172899                                    Austin (TX)  Owner Surrender   
172905                2108 Ferguson Ln in Austin (TX)            Stray   
172903                2108 Ferguson Ln in Austin (TX)            Stray   
172909        Crestview And Park Dr in Jonestown (TX)            Stray   

       Animal Type                        Breed         Color  Age in Months  \
119397         Dog         Spinone Italiano Mix  Yellow/White           72.0   
5647           Dog         Spinone Italiano Mix  Yellow/White           84.0   
26429          Dog         Spinone Italiano Mix  Yellow/White          120.0   
92797          Dog                    Dachshund      Tricolor          120.0   
126267         Dog            Shetland Sheepdog   Brown/White          192.0   
...            ...                          ...           ...            ...   
172901         Dog           Great Pyrenees Mix         White            5.0   
172899         Dog       Labrador Retriever Mix   White/Black           24.0   
172905         Dog      Chihuahua Shorthair Mix      Tricolor            4.0   
172903         Dog      Chihuahua Shorthair Mix      Tricolor            4.0   
172909         Dog  Labrador Retriever/Pit Bull    White/Gray           84.0   

       Reproductive Status  Gender  Condition Category  Entry Index  
119397            Neutered    Male             Healthy            1  
5647              Neutered    Male             Healthy            2  
26429             Neutered    Male             Healthy            3  
92797             Neutered    Male             Healthy            1  
126267            Neutered    Male  Medical - Moderate            1  
...                    ...     ...                 ...          ...  
172901              Intact  Female  Medical - Moderate            1  
172899            Neutered    Male             Healthy            1  
172905              Intact    Male             Healthy            1  
172903              Intact    Male             Healthy            1  
172909             Unknown    None             Healthy            1  

[172911 rows x 13 columns]

Next we will create the dataframe

In [60]:
# Keep only the first outcome per Animal ID
outcomes = outcomes.drop_duplicates(subset="Animal ID", keep="first")

# Now merge
estimated_stay = pd.merge(
    intakes[["Animal ID", "Intake Date", "Entry Index"]],
    outcomes[["Animal ID", "Outcome Date", "Entry Index"]],
    on=["Animal ID", "Entry Index"],
    how="left"
)


In [61]:
estimated_stay.shape

(172911, 4)

In [62]:
estimated_stay.head(20)

Animal ID         Intake Date  Entry Index        Outcome Date
0    A006100 2014-03-07 14:26:00            1 2014-03-08 17:10:00
1    A006100 2014-12-19 10:21:00            2                 NaT
2    A006100 2017-12-07 14:07:00            3                 NaT
3    A047759 2014-04-02 15:55:00            1 2014-04-07 15:12:00
4    A134067 2013-11-16 09:02:00            1 2013-11-16 11:54:00
5    A141142 2013-11-16 14:46:00            1 2013-11-17 11:40:00
6    A163459 2014-11-14 15:11:00            1 2014-11-14 19:28:00
7    A165752 2014-09-15 11:28:00            1 2014-09-15 16:35:00
8    A169438 2018-04-04 20:37:00            1 2018-04-04 20:47:00
9    A178569 2014-03-17 09:45:00            1 2014-03-23 15:57:00
10   A189592 2015-09-18 17:46:00            1 2015-09-18 19:04:00
11   A191351 2015-11-13 15:57:00            1 2015-11-17 13:29:00
12   A197810 2014-12-08 12:30:00            1 2014-12-22 15:23:00
13   A200922 2013-10-03 15:47:00            1 2013-11-22 09:44:00
14   A208755 2015-02-17 13:17:00            1 2015-02-17 16:25:00
15   A210457 2016-09-28 12:05:00            1 2016-10-07 12:34:00
16   A212672 2013-11-25 17:18:00            1 2013-12-06 14:34:00
17   A214991 2013-12-14 11:51:00            1 2013-12-14 13:28:00
18   A215368 2014-08-04 13:51:00            1 2014-08-04 16:31:00
19   A218622 2013-12-02 15:33:00            1 2013-12-02 17:09:00

In [90]:
estimated_stay.info()

<class 'pandas.core.frame.DataFrame'>
Index: 129031 entries, 0 to 172887
Data columns (total 6 columns):
 #   Column         Non-Null Count   Dtype         
---  ------         --------------   -----         
 0   Animal ID      129031 non-null  object        
 1   Intake Date    129031 non-null  datetime64[ns]
 2   Entry Index    129031 non-null  int64         
 3   Outcome Date   129030 non-null  datetime64[ns]
 4   Time Spent     129030 non-null  float64       
 5   Time Category  129031 non-null  object        
dtypes: datetime64[ns](2), float64(1), int64(1), object(2)
memory usage: 10.9+ MB


In [91]:
estimated_stay = estimated_stay.drop_duplicates(subset='Outcome Date', keep='first')

In [92]:
estimated_stay.head()

Animal ID         Intake Date  Entry Index        Outcome Date  Time Spent  \
0   A006100 2014-03-07 14:26:00            1 2014-03-08 17:10:00         1.0   
1   A006100 2014-12-19 10:21:00            2                 NaT         NaN   
3   A047759 2014-04-02 15:55:00            1 2014-04-07 15:12:00         4.0   
4   A134067 2013-11-16 09:02:00            1 2013-11-16 11:54:00         0.0   
5   A141142 2013-11-16 14:46:00            1 2013-11-17 11:40:00         0.0   

        Time Category  
0  Less than 6 months  
1       Did not Leave  
3  Less than 6 months  
4  Less than 6 months  
5  Less than 6 months

In [66]:
estimated_stay["Intake Date"].isnull().sum()

0

In [63]:
estimated_stay.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 172911 entries, 0 to 172910
Data columns (total 3 columns):
 #   Column        Non-Null Count   Dtype         
---  ------        --------------   -----         
 0   Animal ID     172911 non-null  object        
 1   Intake Date   172911 non-null  datetime64[ns]
 2   Outcome Date  172205 non-null  datetime64[ns]
dtypes: datetime64[ns](2), object(1)
memory usage: 4.0+ MB


In [93]:
estimated_stay["Time Spent"] = (estimated_stay["Outcome Date"] - estimated_stay["Intake Date"]).dt.days

In [94]:
estimated_stay.head()

Animal ID         Intake Date  Entry Index        Outcome Date  Time Spent  \
0   A006100 2014-03-07 14:26:00            1 2014-03-08 17:10:00         1.0   
1   A006100 2014-12-19 10:21:00            2                 NaT         NaN   
3   A047759 2014-04-02 15:55:00            1 2014-04-07 15:12:00         4.0   
4   A134067 2013-11-16 09:02:00            1 2013-11-16 11:54:00         0.0   
5   A141142 2013-11-16 14:46:00            1 2013-11-17 11:40:00         0.0   

        Time Category  
0  Less than 6 months  
1       Did not Leave  
3  Less than 6 months  
4  Less than 6 months  
5  Less than 6 months

In [67]:
estimated_stay["Time Category"] = ""  # Create empty column

for index, row in estimated_stay.iterrows():
    s = row["Time Spent"]
    n = row["Outcome Date"]
    
    if pd.isna(n):
        estimated_stay.at[index, "Time Category"] = "Did not Leave"
    elif s < 182:
        estimated_stay.at[index, "Time Category"] = "Less than 6 months"
    elif s < 365:
        estimated_stay.at[index, "Time Category"] = "Less than a year"
    else:
        estimated_stay.at[index, "Time Category"] = "More than a year"

In [68]:
estimated_stay["Time Category"].unique()

array(['Less than 6 months', 'Did not Leave', 'Less than a year',
       'More than a year'], dtype=object)

# Final Check

In [69]:
intakes.head()

Animal ID    Name         Intake Date  \
119397   A006100   Scamp 2014-03-07 14:26:00   
5647     A006100   Scamp 2014-12-19 10:21:00   
26429    A006100   Scamp 2017-12-07 14:07:00   
92797    A047759    Oreo 2014-04-02 15:55:00   
126267   A134067  Bandit 2013-11-16 09:02:00   

                                       Found Location      Intake Type  \
119397                   8700 Research in Austin (TX)    Public Assist   
5647                8700 Research Blvd in Austin (TX)    Public Assist   
26429   Colony Creek And Hunters Trace in Austin (TX)            Stray   
92797                                     Austin (TX)  Owner Surrender   
126267             12034 Research Blvd in Austin (TX)    Public Assist   

       Animal Type                 Breed         Color  Age in Months  \
119397         Dog  Spinone Italiano Mix  Yellow/White           72.0   
5647           Dog  Spinone Italiano Mix  Yellow/White           84.0   
26429          Dog  Spinone Italiano Mix  Yellow/White          120.0   
92797          Dog             Dachshund      Tricolor          120.0   
126267         Dog     Shetland Sheepdog   Brown/White          192.0   

       Reproductive Status Gender  Condition Category  Entry Index  
119397            Neutered   Male             Healthy            1  
5647              Neutered   Male             Healthy            2  
26429             Neutered   Male             Healthy            3  
92797             Neutered   Male             Healthy            1  
126267            Neutered   Male  Medical - Moderate            1

In [70]:
outcomes.head()

Animal ID    Name        Outcome Date MonthYear Date of Birth  \
150491   A006100   Scamp 2014-03-08 17:10:00    Mar-14    2007-07-09   
51216    A047759    Oreo 2014-04-07 15:12:00    Apr-14    2004-04-02   
106445   A134067  Bandit 2013-11-16 11:54:00    Nov-13    1997-10-16   
168829   A141142  Bettie 2013-11-17 11:40:00    Nov-13    1998-06-01   
172531   A163459   Sasha 2014-11-14 19:28:00    Nov-14    1999-10-19   

           Outcome Type Animal Type Sex upon Outcome  \
150491  Return to Owner         Dog    Neutered Male   
51216          Transfer         Dog    Neutered Male   
106445  Return to Owner         Dog    Neutered Male   
168829  Return to Owner         Dog    Spayed Female   
172531  Return to Owner         Dog    Intact Female   

                              Breed         Color  Age in Months  \
150491         Spinone Italiano Mix  Yellow/White      79.960578   
51216                     Dachshund      Tricolor     120.137976   
106445            Shetland Sheepdog   Brown/White     193.002628   
168829  Labrador Retriever/Pit Bull   Black/White     185.545335   
172531      Miniature Schnauzer Mix    Black/Gray     180.847569   

       Reproductive Status  Gender Outcome Category  Entry Index  
150491            Neutered    Male         Positive            1  
51216             Neutered    Male         Positive            1  
106445            Neutered    Male         Positive            1  
168829              Spayed  Female         Positive            1  
172531              Intact  Female         Positive            1

In [71]:
estimated_stay.head()

Animal ID         Intake Date  Entry Index        Outcome Date  Time Spent  \
0   A006100 2014-03-07 14:26:00            1 2014-03-08 17:10:00         1.0   
1   A006100 2014-12-19 10:21:00            2                 NaT         NaN   
3   A047759 2014-04-02 15:55:00            1 2014-04-07 15:12:00         4.0   
4   A134067 2013-11-16 09:02:00            1 2013-11-16 11:54:00         0.0   
5   A141142 2013-11-16 14:46:00            1 2013-11-17 11:40:00         0.0   

        Time Category  
0  Less than 6 months  
1       Did not Leave  
3  Less than 6 months  
4  Less than 6 months  
5  Less than 6 months

In [86]:
#Save the files as csv
intakes.to_csv("Intakes_new.csv")

In [87]:
outcomes.to_csv("Outcomes_new.csv")

In [88]:
estimated_stay.to_csv("Estimated_Stay.csv")

In [89]:
DOB.to_csv("Date_of_birth.csv")

## Analysis and Findings

The first dashboard provides a basic yet informative overview of the animal shelter’s activity and performance. With a total of approximately 172,000 animals, the majority were taken in due to abandonment, while most left through adoption—highlighting the shelter’s overall success in securing positive outcomes. About 71% of animals had a favorable resolution, though the remaining 29% faced less desirable outcomes, suggesting areas for further attention. The average length of stay is around one month, indicating good turnover, although dogs tend to stay longer than other animals. Cats and dogs dominate the shelter population, with birds, livestock, and wildlife under the “Other” category being less frequent. Trends over the past 13 years show a sharp drop in intake and outcome activity after 2019, likely due to the COVID-19 pandemic. This dashboard serves as a high-level summary; deeper analysis into factors such as animal characteristics, health conditions, and seasonal trends will be explored in the subsequent dashboards.

The second dashboard explores how intake type influences animal outcomes, revealing that most animals experience positive outcomes—primarily through adoption, return to owner (RTO), or transfer—regardless of how they entered the shelter. This suggests that the shelter effectively facilitates favorable results even for animals arriving under challenging circumstances such as abandonment or as strays. Additionally, the map highlights the geographic distribution of intake locations, with most animals being found within Austin City. However, a few notable outliers appear around Dallas, indicating either a broader reach of the shelter’s services or occasional transfers from more distant areas. This spatial insight can help the shelter evaluate its operational reach and target community outreach or rescue efforts more effectively.

The third dashboard provides insights into how animal characteristics—such as color, type, and reproductive status—affect their length of stay in the shelter. One key observation is that animals that have undergone spaying or neutering tend to stay longer than those that are intact. This could suggest that adopters may prefer intact animals, possibly due to ethical concerns about surgical procedures or because they intend to breed them. Additionally, a pattern emerges regarding color: darker-colored animals generally experience shorter shelter stays compared to lighter-colored ones. This trend may reflect adopter preferences or subconscious biases related to animal appearance. Overall, these findings highlight how certain traits may influence adoptability and can inform future shelter practices and marketing strategies.


The final dashboard highlights seasonal trends in shelter activity, revealing clear patterns in both intakes and outcomes throughout the year. Notably, the month of May experiences a spike in animal intakes that surpasses the number of outcomes, suggesting a seasonal influx—possibly due to breeding cycles or increased stray reports in warmer months. In contrast, for the rest of the year, outcomes consistently outpace intakes, indicating efficient shelter operations. Across all months, adoption remains the most common outcome, reinforcing its role as the primary path to positive results. These trends can help the shelter anticipate resource needs and plan adoption campaigns during peak intake periods.


## Recommendations

#### 1-Prepare for Seasonal Spikes (Especially in May):
Since intake rates peak in May, the shelter should allocate more resources—such as staff, medical supplies, and foster home availability—during this period. Proactive adoption campaigns and partnerships with rescue organizations during the spring could help manage the surge.

#### 2-Targeted Adoption Campaigns for Less Adopted Animals:
Animals with longer stays—such as those that are neutered/spayed or of lighter colors—may benefit from dedicated marketing efforts. These could include promotional events, social media stories, or reduced adoption fees to increase visibility and appeal.

#### 3-Leverage Predictive Modeling to Improve Outcomes:
Implement decision or decomposition tree models to assess the likelihood of positive outcomes based on intake characteristics (age, condition, color, reproductive status). This can support staff in prioritizing intervention and care strategies.

#### 4-Focus on Intact Animal Policies:
The trend of intact animals being adopted faster suggests the need for clear educational outreach. Educating the public on the health and community benefits of spaying/neutering could balance adoption rates without compromising animal welfare goals.

#### 5-Strengthen Stray Animal Management in Surrounding Areas:
Since animals are found as far as Dallas, collaborating with neighboring cities or establishing satellite intake points might improve response times and reduce animal stress during transport.

#### 6-Continue Efficient Shelter Management:
The low average length of stay and high rate of positive outcomes indicate strong operational performance. Maintaining or improving this standard will require ongoing monitoring and refinement of processes based on data insights.

## Limitations and Assumptions

#### 1 - Excessive Granularity
The dataset contained an overly detailed breakdown of variables such as animal color, breed, and intake condition. While such granularity can be valuable in specialized studies, it posed challenges for conducting a broad-level analysis. Simplifying and grouping these variables into more readable and comparable categories proved to be time-consuming.

#### 2 - Inconsistent Age Format
The age data lacked consistency, with values recorded in a mix of weeks, months, and years. This lack of uniformity introduced complexity in data preparation and required standardization to a common unit to allow accurate analysis.

#### 3 - Absence of Length of Stay Column
A direct measure of each animal’s length of stay in the shelter was not provided in the dataset. As a result, this metric had to be manually derived by calculating the difference between intake and outcome dates.

#### 4 - Incomplete Data
Despite recording intake and outcome dates, several entries were missing the animal's date of birth. This created additional challenges during data cleaning and transformation, requiring careful handling to ensure the integrity of existing information was maintained.


## References

https://data.austintexas.gov/browse?q=animal&sortBy=relevance&pageSize=20&page=1